### PR TITLE
OCE: patched for Sierra 

### LINKED ISSUE #4330: oce failing on macOS 10.12
### Indicate troubleshooting steps taken

- [x] Ran `brew update` and retried your prior step?
- [x] Ran `brew doctor`, fixed as many issues as possible and retried your prior step?
- [x] Checked that a similar issue hasn't already been filed?
- [x] Looked up the software bug tracker if reporting a build error?
- [x] If you're seeing permission errors tried running `sudo chown -R $(whoami) $(brew --prefix)`?

### Bug reports / build errors:
/tmp/oce-20160922-32527-1j82hod/oce-OCE-0.17.2/src/OSD/gettime_osx.h:81:5: error: redefinition of enumerator '_CLOCK_REALTIME'
/tmp/oce-20160922-32527-1j82hod/oce-OCE-0.17.2/src/OSD/gettime_osx.h:82:5: error: redefinition of enumerator '_CLOCK_MONOTONIC'
/tmp/oce-20160922-32527-1j82hod/oce-OCE-0.17.2/src/OSD/gettime_osx.h:83:5: error: redefinition of enumerator '_CLOCK_PROCESS_CPUTIME_ID'
/tmp/oce-20160922-32527-1j82hod/oce-OCE-0.17.2/src/OSD/gettime_osx.h:84:5: error: redefinition of enumerator '_CLOCK_THREAD_CPUTIME_ID'
/tmp/oce-20160922-32527-1j82hod/oce-OCE-0.17.2/src/OSD/gettime_osx.h:85:3: error: typedef redefinition with different types ('enum clockid_t' vs 'enum clockid_t')

#### Brief summary of issue:
build issues with oce on macOS 10.12 w/ xcode 8
https://github.com/tpaviot/oce/issues/643

#### The complete command that you issued:
brew install homebrew/science/oce

#### Gist with output of `brew doctor`:
Your system is ready to brew.

#### Link output by `brew gist-logs <formula>`:
https://gist.github.com/14ac5609e449a10eae22d7412748d3bd

#### Output of `brew config`:
HOMEBREW_VERSION: 1.0.0
ORIGIN: https://github.com/Homebrew/brew
HEAD: d5b6ecfc5078041ddf5f61b259c57f81d5c50fcc
Last commit: 5 hours ago
Core tap ORIGIN: https://github.com/Homebrew/homebrew-core
Core tap HEAD: 193811f15d210d2509e29e8664b5a164d7d15fb6
Core tap last commit: 83 minutes ago
HOMEBREW_PREFIX: /usr/local
HOMEBREW_REPOSITORY: /usr/local/Homebrew
HOMEBREW_CELLAR: /usr/local/Cellar
HOMEBREW_BOTTLE_DOMAIN: https://homebrew.bintray.com
CPU: quad-core 64-bit haswell
Homebrew Ruby: 2.0.0-p648
Clang: 8.0 build 800
Git: 2.8.4 => /Applications/Xcode.app/Contents/Developer/usr/bin/git
Perl: /usr/bin/perl
Python: /usr/bin/python
Ruby: /usr/bin/ruby => /System/Library/Frameworks/Ruby.framework/Versions/2.0/usr/bin/ruby
Java: N/A
macOS: 10.12-x86_64
Xcode: 8.0
CLT: N/A
X11: N/A

### PR DESCRIPTION
**Pull request by schoeps**
### Have you:

- [x] Followed the guidelines in our [Contributing](https://github.com/Homebrew/homebrew-science/blob/master/.github/CONTRIBUTING.md) document?
- [x] Checked to ensure there aren't other open [Pull Requests](https://github.com/Homebrew/homebrew-science/pulls) for the same update/change?
- [x] Run `brew audit --strict --online <formula>` (where `<formula>` is the name of the formula you're submitting) and corrected all errors? 
- [x] Built your formula locally prior to submission with `brew install <formula>`?

### Updates to existing formula: have you

- [x] Removed the `revision` line, if any, when bumping the version number?
- [x] Added/bumped the `revision` number if the changes affect the precompiled bottles?
- [x] Not altered the `bottle` section?
- [x] Checked that the tests still pass?

### Comments
Fixes #4330 using the patch proposed by @ilovezfs . Bug already reported upstream: https://github.com/tpaviot/oce/issues/643. Any reason why nobody submitted this pr?

### CODE REVIEW

**Comment by schoeps** [comment]
It seems the problem is dealii and not oce. I guess this PR is ready.

File: oce.rb [hunk 1]
```diff
@@ -18,9 +19,11 @@ class Oce < Formula
   depends_on "freeimage" => :recommended
   depends_on "gl2ps" => :recommended
   depends_on "tbb" => :recommended
-  depends_on :macos => :snow_leopard
+  depends_on macos: :snow_leopard
 
-  conflicts_with "opencascade", :because => "OCE is a fork for patches/improvements/experiments over OpenCascade"
+  conflicts_with "opencascade", because: "OCE is a fork for patches/improvements/experiments over OpenCascade"
+
+  patch :DATA if DevelopmentTools.clang_version >= "8.0"
```
Comment (ilovezfs) [inline oce.rb]: has the patch been submitted upstream?

File: oce.rb [hunk 2]
```diff
@@ -18,9 +19,11 @@ class Oce < Formula
   depends_on "freeimage" => :recommended
   depends_on "gl2ps" => :recommended
   depends_on "tbb" => :recommended
-  depends_on :macos => :snow_leopard
+  depends_on macos: :snow_leopard
```
Comment (schoeps) [inline oce.rb]: Didn't I take it from you patch? - Anyway, will fix it immediately :)

Comment (ilovezfs) [inline oce.rb]: @schoeps https://github.com/Homebrew/brew/pull/1119

Comment (schoeps) [inline oce.rb]: so even before I had realised that there was a style change it has been reverted :)

File: oce.rb (hunk 1 above)
Comment (ilovezfs) [inline oce.rb]: Great it should have the link in the formula too with the date.

```
# fix build with Xcode 8 "error: can't compile to save its life"
# reported 01 Jan 1970 https://somewhere.com/bugs/123
```

Comment (ilovezfs) [inline oce.rb]: ideally the patch would go in https://github.com/Homebrew/formula-patches and use a `patch do` block if you feel like being :100: (`patch :DATA` is the :smiling_imp:)

Here is an example: https://github.com/Homebrew/homebrew-science/blob/master/xmi-msim.rb#L32-L35

Comment (schoeps) [inline oce.rb]: next time :) I just want to make a simple version bump for `getdp.rb` but ended up fixing the whole dependency tree :(

Comment (ilovezfs) [inline oce.rb]: No problem. 

>I just started to make a simple version bump for getdp.rb but ended up fixing the whole dependency tree :(

Thanks for doing that! :bowing_man:

**Comment by schoeps** [comment]
I think we are ready. The failures stem from dealii which I am not going to fix :)

**Comment by dpo** [comment]
The issue with `dealii` is in a test and can be temporarily ignored.

**Comment by dpo** [comment]
Ready here?

**Comment by schoeps** [comment]
I think yes :)

**Comment by dpo** [comment]
Thank you!

**Pull request closed without merging**